### DISCUSSION TITLE
Add support for variant stacking

### DISCUSSION THREAD

**Discussion by adamwathan** [tailwindlabs/tailwindcss#1373] [Ideas]
Had an idea for syntax that would work for this, so wanted to capture here so I don’t forget:

```js
variants: {
  boxShadow: [
    { responsive: [‘hover’, ‘focus’] },
}
```

I see this being useful if we wanted to add a “dark mode” variant for example and people wanted to do stuff like “md:dark:hover:bg-red-500” .

**Comment by haversnail** [comment]
@adamwathan, would very much love to see this. 🙌 Currently writing a dark-variant plugin myself that leverages the `prefers-color-scheme` media feature, but truth be told I didn't realize you couldn't stack the variant prefixes like that until I came across this issue.

Btw, absolutely love the framework — thanks for making this a quality piece of work! 🎉

**Comment by kn0wn** [comment]
@adamwathan this would be great to have implemented, I have a use case where I need to use both group-hover and group-focus having duplicate inline css just looks messy
